src/python/genny/cedar.py

```diff
@@ -0,0 +1,204 @@
+    SIZE = 7
+    WORKERS = 8
+
+
+class IntermediateCSVReader:
+    """
+    Class that reads IntermediateCSV and outputs Cedar BSON
+    """
+
+    def __init__(self, reader):
+        self.raw_reader = reader
+        self.cum_vals = [0 for _ in range(9)]
+        self.prev_ts = 0
+
+    def __iter__(self):
+        return self
+
+    def __next__(self):
+        return self._parse_into_cedar_format(next(self.raw_reader))
+
+    def _parse_into_cedar_format(self, line):
+        # Compute all cumulative values for simplicity; Not all values are used.
+        self.cum_vals = [sum(v) for v in zip(line, self.cum_vals)]
+
+        res = OrderedDict([
+            ('ts', line[IntermediateCSV.TS_MS]),
+            ('id', line[IntermediateCSV.THREAD]),
+            ('counters', OrderedDict([
+                ('n', self.cum_vals[IntermediateCSV.N]),
+                ('ops', self.cum_vals[IntermediateCSV.OPS]),
+                ('size', self.cum_vals[IntermediateCSV.SIZE]),
+                ('errors', self.cum_vals[IntermediateCSV.ERRORS])
+            ])),
+            ('timers', OrderedDict([
+                ('duration', self.cum_vals[IntermediateCSV.DURATION]),
+                ('total', 1)  # FIXME: compute total
+            ])),
+            ('gauges', OrderedDict([
+                ('workers', line[IntermediateCSV.WORKERS])
+            ]))
+        ])
+
+        self.prev_ts = line[IntermediateCSV.TS_MS]
+
+        return res
+
+
+def compute_cumulative_and_write_to_bson(file_name, out_dir):
+    # Remove ".csv" and add ".bson"
+    out_file_name = file_name[:-4] + '.bson'
+    with open(pjoin(out_dir, out_file_name), 'wb') as out_f, open(pjoin(out_dir, file_name)) as in_f:
+        for ordered_dict in IntermediateCSVReader(csv.reader(in_f, quoting=csv.QUOTE_NONNUMERIC)):
+            out_f.write(BSON.encode(ordered_dict))
+
+
+def split_into_actor_operation_csv_files(data_reader, out_dir):
+    """
+    Split up the monolithic genny metrics csv2 file into smaller [actor]-[operation].csv files
+    """
+    cur_out_csv = None
+    cur_out_fh = None
+    cur_actor_op_pair = (None, None)
+    output_files = []
+
+    for line, actor, op in data_reader:
+        if (actor, op) != cur_actor_op_pair:
+            cur_actor_op_pair = (actor, op)
+
+            # Close out old file.
+            if cur_out_fh:
+                cur_out_fh.close()
+
+            # Open new csv file.
+            file_name = actor + '-' + op + '.csv'
+            output_files.append(file_name)
+            cur_out_fh = open(pjoin(out_dir, file_name), 'w', newline='')
+
+            # Quote non-numeric values so they get converted to float automatically
+            cur_out_csv = csv.writer(cur_out_fh, quoting=csv.QUOTE_NONNUMERIC)
+
+        cur_out_csv.writerow(line)
+
+    cur_out_fh.close()
+
+    return output_files
+
+
+def sort_csv_file(file_name, out_dir):
+    # Sort on Timestamp and Thread.
+    csvsort(pjoin(out_dir, file_name),
+            [0, 1],
```

Comment:
0 and 1 seem magical (although I know probably copying from the ticket). Can we use `_Columns` and/or do this based on the header row values?